### ISSUE TITLE
0.13: [BUG] Terraform Deploy action type error

### ISSUE THREAD

**Issue by worldofgeese** [garden-io/garden#4266]
@edvald in that case looks like I may have run into a bug converting the provider to an action

```
kind: Deploy
type: terraform
name: my-terraform
spec: 
  autoApply: true
  variables:
    project_id: devrel-348008
```

```shell
✖ deploy.my-terraform  → 
Failed processing resolve Deploy type=terraform name=my-terraform. Here is the output:
────────────────────────────────────────────────────────────────────────────────
Could not find type definition for Deploy type=terraform name=my-terraform.
This is a bug. Please report it!
──────────────────────────────────────────────────────────────────────────────── (in 0 sec)
ℹ deploy.my-mongodb    → Already deployed
✖ 1 deploy action(s) failed!
```

_Originally posted by @worldofgeese in https://github.com/garden-io/garden/issues/3475#issuecomment-1544184859_

cc @Orzelius

**Comment by Orzelius** [comment]
yes this is a bug, looking into it asap

**Comment by worldofgeese** [comment]
@Orzelius nope: I removed it because I thought it was an either a provider level configuration or action-level configuration due to the duplicative functionality. Should I have just a one-string configuration to enable TF e.g.

```
providers:
  - # The name of the provider plugin to use.
    name: local-kubernetes

    context: minikube

  - name: terraform
 ```

?

**Comment by Orzelius** [comment]
The project level provider config tells which plugins garden should initialize so it's expected that terraform actions don't work if the plugin is not configured. You can have the configuration as just `name: terraform` if you don't need to specify anything else.

You can keep this issue open as the error is quite misleading and needs some work

**Comment by worldofgeese** [comment]
[Here's ](https://gist.github.com/worldofgeese/0c5bac7bb16bbc829a6eb62592ddef1f)everything you'd need to duplicate my environment

**Comment by Orzelius** [comment]
closed via  #4272
